From 47a56f0d49b13f605cb55f19c0ec515cb91d96c6 Mon Sep 17 00:00:00 2001
From: Samuel Cabrero <kernevil@sefirot.kernevil.me>
Date: Sun, 5 Feb 2017 12:29:47 +0100
Subject: [PATCH] Make gstimxcommon library public

Necessary when implementing external sinks to retrieve the mapped
physical memory address from gstreamer buffers.
---
 src/common/gstimxcommon.pc.in | 10 ++++++++++
 src/common/wscript            | 12 ++++++++++++
 2 files changed, 22 insertions(+)
 create mode 100644 src/common/gstimxcommon.pc.in

diff --git a/src/common/gstimxcommon.pc.in b/src/common/gstimxcommon.pc.in
new file mode 100644
index 0000000..5f06312
--- /dev/null
+++ b/src/common/gstimxcommon.pc.in
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=${prefix}
+includedir=${prefix}/include
+libdir=${exec_prefix}/lib
+
+Name: gstimxcommon
+Description: GStreamer IMX memory allocators
+Version: @VERSION@
+Cflags: -I${includedir}/gstreamer-1.0/gst/allocators/imx @CFLAGS@
+Libs: -L${libdir} @LIBS@
diff --git a/src/common/wscript b/src/common/wscript
index 22b2815..98d4941 100644
--- a/src/common/wscript
+++ b/src/common/wscript
@@ -17,3 +17,15 @@ def build(bld):
 	)
 	if bld.env['BUILD_FOR_ANDROID']:
 		obj.install_path = os.path.join(bld.env['PREFIX'], 'lib')
+
+	bld(
+		source = 'gstimxcommon.pc.in',
+		target = 'gstimxcommon.pc',
+		VERSION = bld.env['GSTIMX_VERSION'],
+        LIBS = '-lgstimxcommon',
+		CFLAGS = '-Wall'
+	)
+
+	bld.install_files('${PREFIX}/include/gstreamer-1.0/gst/allocators/imx', 'phys_mem_allocator.h')
+	bld.install_files('${PREFIX}/include/gstreamer-1.0/gst/allocators/imx', 'phys_mem_addr.h')
+	bld.install_files('${PREFIX}/include/gstreamer-1.0/gst/allocators/imx', 'phys_mem_meta.h')
